### PR TITLE
Init Project

### PR DESCRIPTION
**Pull request by hassan-aizuddin**
Ready

### CODE REVIEW

**Comment by hassan-aizuddin** [comment]
Ready

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+source 'https://rubygems.org'
```
Comment (chungpv-1008) [inline Gemfile]: remove all comments, using " instead of ' in rb, erb files

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// This file is automatically compiled by Webpack, along with any other files
+// present in this directory. You're encouraged to place your actual application logic in
+// a relevant structure within app/javascript and only use these pack files to reference
+// that code so it'll be compiled.
+
+require("@rails/ujs").start()
+require("turbolinks").start()
+require("@rails/activestorage").start()
+require("channels")
```
Comment (chungpv-1008) [inline app/javascript/packs/application.js]: using ' instead of " in js file

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>SampleApp</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Comment (chungpv-1008) [inline app/views/layouts/application.html.erb]: using " instead of '

File: config/application.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require_relative 'boot'
+
+require 'rails/all'
+
+# Require the gems listed in Gemfile, including any gems
```
Comment (chungpv-1008) [inline config/application.rb]: using " instead of '

File: config/credentials.yml.enc [hunk 5]
```diff
@@ -0,0 +1 @@
+H7Qek//UK0uqK0ZIMLWt69f9Fr9OvDDc6VDK3UGXSDJst+WjMYeYUk3dLqFRFkAPkqnE/vub9E0OaIu4mnBKF4rfm/DTh4UMWiftIWXBiNzPCqI51a3e7dz4JLuM/vraNucg5akpX6cDlKyPPg/XQfZKOTG5yqXAXkCd76oZxPNSnZJFTOG3JujXKtRARLpdaNUXSHZi+adlN6EX6aZxAOMFoJd9LH8oW/Tpo9tu6tuGgFiekay0NWJbGOW2r5GOOmOuB94e/ymaANsioueTdiBx50He4EomdO6sXCWnVq0IpbaSe5PQhjeTeHBN2/vPAjsdYL6QvCddFDF13yN12JK3a3E21iw3JOmcUBtZeJCmO/3YnScw6a0draLUYntN6man/lUrkUbqQtK3ThksHj4M398uwB1U2gVm--/iKZ1yXswZ9t8XiU--+KxeFwIENpp9dvtoerRp3A==
```
Comment (chungpv-1008) [inline config/credentials.yml.enc]: ignore this file in file '.gitignore'. Create empty file 'credentials.yml.enc.example'

File: config/database.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  database: db/development.sqlite3
+
+# Warning: The database defined as "test" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: db/test.sqlite3
+
+production:
```
Comment (chungpv-1008) [inline config/database.yml]: install 'mysql2' gem, using 'mysql2' instead of 'sqlite3'

File: config/routes.rb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
```
Comment (chungpv-1008) [inline config/routes.rb]: remove all comments

File: test/application_system_test_case.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
```
Comment (chungpv-1008) [inline test/application_system_test_case.rb]: ignore folder 'test' in file '.gitignore'. You don't need follow 'test' in turtorial anymore

**Comment by hassan-aizuddin** [comment]
Ready

File: .gitignore [hunk 9]
```diff
@@ -1,56 +1,43 @@
...
+/config/master.key
+
+/public/packs
+/public/packs-test
+/node_modules
+/yarn-error.log
+yarn-debug.log*
+.yarn-integrity
+/config/credentials.yml.enc 
+/test/application_system_test_case.rb
```
Comment (saobv-2223) [inline .gitignore:43]: image

File: app/mailers/application_mailer.rb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
+  layout 'mailer'
```
Comment (saobv-2223) [inline app/mailers/application_mailer.rb:3]: image

File: config/database.yml [hunk 11]
```diff
@@ -0,0 +1,54 @@
+# MySQL. Versions 5.5.8 and up are supported.
```
Comment (saobv-2223) [inline config/database.yml]: image

File: test/channels/application_cable/connection_test.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+require "test_helper"
```
Comment (saobv-2223) [inline test/channels/application_cable/connection_test.rb:1]: image

**Comment by hassan-aizuddin** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**